src/data_conn.c

```
@@ -293,6 +293,8 @@ handle_data_conn_error(replica_t *r)
 
 	MTX_LOCK(&r->r_mtx);
 	if (epoll_ctl(r->epollfd, EPOLL_CTL_DEL, r->iofd, NULL) == -1) {
+		shutdown(r->iofd, SHUT_RDWR);
```

If epoll_ctl system call got an error, it implies that something weird happened in the system. In such case better to get rid-off this replica. Why we do by returning from here ?

---

yes. This is replica removal path from the target, So we can ignore error here.